### ISSUE TITLE
Confusing error messages with reg exp

### ISSUE THREAD

**Issue by Ana06** [bkuhlmann/git-cop#22]
I have the following code:

```
:commit_subject_prefix:
  :enabled: true
  :whitelist:
    - \A(?!\s)(\[api\])?(\[backend\])?(\[ci\])?(\[dist\])?(\[doc\])?(\[webui\])? [a-zA-Z]
```

When writing a commit that doesn't match the regular expression I get an error like:

```
4a85f0bd18443e326db6fc133df34b8963fa8466 (Ana María Martínez Gómez, 2 weeks ago): [webui][ci] Run git-cop in Travis
  ERROR: Commit Subject Prefix. Invalid prefix. Use: "A(?!s)([api])?([backend])?([ci])?([dist])?([doc])?([webui])? [a-zA-Z]".
```

Both the error message and the way the regular expression are render are confusing. :confounded: I would say the error message should say something like: `ERROR: Commit Subject Prefix. Invalid prefix. It should match the regular expression: ` And the regular expression should be rendered without escaping the `\`.

**Comment by bkuhlmann** [comment]
I think the thing to do here is detect whether the value is string or regex and then print them out in the appropriate format because the graylist can be a mixed bag of strings and/or regular expressions.

**Comment by Ana06** [comment]
Yes, I was also thinking that the fact of being able to mixture string and regular expressions is a little bit confusing. And also, currently because of the implementation everything is considered to be a regular expression. Maybe we can have something to indicate if it is a reg expr or no, different syntax in the whitelist or something like that.
